src/kani-compiler/src/codegen_cprover_gotoc/codegen/place.rs

```diff
@@ -138,13 +138,13 @@ impl<'tcx> ProjectedPlace<'tcx> {
         }
     }
 
-    pub fn new(
+    pub fn try_new(
         goto_expr: Expr,
         mir_typ_or_variant: TypeOrVariant<'tcx>,
         fat_ptr_goto_expr: Option<Expr>,
         fat_ptr_mir_typ: Option<Ty<'tcx>>,
         ctx: &mut GotocCtx<'tcx>,
-    ) -> Self {
+    ) -> Result<Self, UnimplementedData> {
         let mir_typ_or_variant = mir_typ_or_variant.monomorphize(ctx);
         let fat_ptr_mir_typ = fat_ptr_mir_typ.map(|t| ctx.monomorphize(t));
         if let Some(fat_ptr) = &fat_ptr_goto_expr {
```

Comment:
Should this also be a return error?

Reply:
Not sure where you are pointing at.

Reply:
We have a mix of `panic` and `Error` in the same function. Was wondering if we should standardize on `Error`

Reply:
I don't think so. Panic is stronger and we should keep doing that if something is really unexpected.

That said, I would be OK compromising and turning those things into debug_assert!() for now so they at least fail in our regression but in the user code, it results in a unsupported_construct.

In the long run, I don't like pushing errors downstream and we should provide stronger guarantees that our compiler works. Everything in this function should eventually trigger an internal compiler error.